evalml/tests/component_tests/test_oversamplers.py

```diff
@@ -0,0 +1,222 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from evalml.pipelines.components import (
+    SMOTENCSampler,
+    SMOTENSampler,
+    SMOTESampler
+)
+
+im = pytest.importorskip('imblearn.over_sampling', reason='Skipping test because imbalanced-learn not installed')
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler, SMOTENCSampler, SMOTENSampler])
+def test_init(sampler):
+    parameters = {
+        "sampling_ratio": 0.5,
+        "sampling_ratio_dict": {},
+        "k_neighbors": 2,
+        "n_jobs": -1
+    }
+    if 'SMOTENC' in sampler.name:
+        parameters['categorical_features'] = [0]
+    oversampler = sampler(**parameters)
+    assert oversampler.parameters == parameters
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler(sampling_ratio=1),
+                                     SMOTENCSampler(sampling_ratio=1, categorical_features=[0]),
+                                     SMOTENSampler(sampling_ratio=1)])
+def test_none_y(sampler):
+    X = pd.DataFrame({"a": [i for i in range(5)],
+                      "b": [1 for i in range(5)]})
+    oversampler = sampler
+    with pytest.raises(ValueError, match="y cannot be none"):
+        oversampler.fit(X, None)
+    with pytest.raises(ValueError, match="y cannot be none"):
+        oversampler.fit_transform(X, None)
+    oversampler.fit(X, pd.Series([0] * 4 + [1]))
+    oversampler.transform(X, None)
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler(sampling_ratio=1),
+                                     SMOTENCSampler(sampling_ratio=1, categorical_features=[0]),
+                                     SMOTENSampler(sampling_ratio=1)])
+@pytest.mark.parametrize("data_type", ["np", "pd", "ww"])
+def test_no_oversample(data_type, sampler, make_data_type, X_y_binary):
+    X, y = X_y_binary
+    X = make_data_type(data_type, X)
+    y = make_data_type(data_type, y)
+
+    oversampler = sampler
+    new_X, new_y = oversampler.fit_transform(X, y)
+
+    if data_type == "ww":
+        X = X.to_dataframe().values
+        y = y.to_series().values
+    elif data_type == "pd":
+        X = X.values
+        y = y.values
+
+    np.testing.assert_equal(X, new_X.to_dataframe().values)
+    np.testing.assert_equal(y, new_y.to_series().values)
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler(sampling_ratio=1),
+                                     SMOTENCSampler(sampling_ratio=1, categorical_features=[1]),
+                                     SMOTENSampler(sampling_ratio=1)])
+@pytest.mark.parametrize("data_type", ["np", "pd", "ww"])
+def test_oversample_imbalanced_binary(data_type, sampler, make_data_type):
```

Comment:
I think its worth adding a test case for `sampling_ratio_dict` given that it takes precedence over `sampling_ratio`

Reply:
Yep, that test is in `test_sampler_dic_overrides_ratio`

Reply:
Thanks for pointing that out!